src/main/java/org/breedinginsight/brapi/v2/dao/BrAPIGermplasmDAO.java

```diff
@@ -157,21 +157,25 @@ private Map<String,BrAPIGermplasm> processGermplasmForDisplay(List<BrAPIGermplas
 
                 String newPedigreeString = "";
                 String namePedigreeString = "";
+                String uuidPedigreeString = "";
                 List<String> parents = Arrays.asList(germplasm.getPedigree().split("/"));
                 if (parents.size() >= 1) {
                     if (programGermplasmByFullName.containsKey(parents.get(0))) {
                         newPedigreeString = programGermplasmByFullName.get(parents.get(0)).getAccessionNumber();
                         namePedigreeString = programGermplasmByFullName.get(parents.get(0)).getDefaultDisplayName();
+                        uuidPedigreeString = programGermplasmByFullName.get(parents.get(0)).getExternalReferences().stream().filter(ref -> ref.getReferenceSource().equals(referenceSource)).map(ref -> ref.getReferenceID()).findFirst().get();
```

Comment:
Can you split this into multiple lines? It is hard to read when chained on a one liner.

Reply:
Also, what happens if the `filter` returns no results. Will your `findFirst` error out?

Reply:
Split into multiple lines, and good point! Switched to using an orElse clause to set the value to “” if nothing is found, which leads to a GID with no link